Assets/__Scripts/Beatmap/Animations/TrackAnimator.cs

```diff
@@ -0,0 +1,155 @@
+                {
+                    prop.UpdateProperty(time);
+                }
+            }
+        }
+
+        public void OnChildrenChanged()
+        {
+            cachedChildren = children.Where(o => o.enabled).ToArray();
+            enabled = cachedChildren.Length > 0;
+            if (animator != null) animator.enabled = enabled;
+            parents.ForEach((t) => t.OnChildrenChanged());
+        }
+
+        private void AddPointDef(IPointDefinition.UntypedParams p, string key)
+        {
+            switch (key)
+            {
+            case "_dissolve":
+            case "dissolve":
+                AddPointDef<float>((ObjectAnimator animator, float f) => animator.Opacity.Add(f), PointDataParsers.ParseFloat, p, 1);
+                break;
+            case "_dissolveArrow":
+            case "dissolveArrow":
+                AddPointDef<float>((ObjectAnimator animator, float f) => animator.OpacityArrow.Add(f), PointDataParsers.ParseFloat, p, 1);
+                break;
+            case "_localRotation":
+            case "localRotation":
+                AddPointDef<Quaternion>((ObjectAnimator animator, Quaternion v) => animator.LocalRotation.Add(v), PointDataParsers.ParseQuaternion, p, Quaternion.identity);
+                break;
+            case "_rotation":
+            case "rotation":
+            case "offsetWorldRotation":
+                AddPointDef<Quaternion>((ObjectAnimator animator, Quaternion v) => animator.WorldRotation.Add(v), PointDataParsers.ParseQuaternion, p, Quaternion.identity);
+                break;
+            case "_position":
+                AddPointDef<Vector3>((ObjectAnimator animator, Vector3 v) => animator.OffsetPosition.Add(v), PointDataParsers.ParseVector3, p, Vector3.zero);
+                break;
+            case "offsetPosition":
+            case "localPosition":
+                AddPointDef<Vector3>((ObjectAnimator animator, Vector3 v) => animator.OffsetPosition.Add(v * 1.667f), PointDataParsers.ParseVector3, p, Vector3.zero);
+                break;
+            case "position":
+                AddPointDef<Vector3>((ObjectAnimator animator, Vector3 v) => animator.WorldPosition.Add(v * 1.667f), PointDataParsers.ParseVector3, p, Vector3.zero);
+                break;
+            case "_scale":
+            case "scale":
+                AddPointDef<Vector3>((ObjectAnimator animator, Vector3 v) => animator.Scale.Add(v), PointDataParsers.ParseVector3, p, Vector3.one);
+                break;
+            case "_color":
+            case "color":
+                AddPointDef<Color>((ObjectAnimator animator, Color v) => animator.Colors.Add(v), PointDataParsers.ParseColor, p, Color.white);
+                break;
+            case "_time":
+            case "time":
+                AddPointDef<float>((ObjectAnimator animator, float f) => animator.SetLifeTime(f), PointDataParsers.ParseFloat, p, -1);
+                break;
+            }
+        }
```

Comment:
considering this seems to share some similarities to `ObjectAnimator.cs`, it may be wise to abstract common functionality into a base `GenericAnimator.cs` class.